packages/material-ui/src/Tabs/Tabs.test.js

```diff
@@ -435,87 +438,103 @@ describe('<Tabs />', () => {
     });
 
     it('should render scroll buttons automatically', () => {
-      const wrapper = shallow(
+      const wrapper = mount(
         <Tabs width="md" onChange={noop} value={0} variant="scrollable" scrollButtons="auto">
           <Tab />
           <Tab />
         </Tabs>,
       );
-      assert.strictEqual(wrapper.find(TabScrollButton).length, 2, 'should be two');
-    });
-
-    it('should should not render scroll buttons automatically', () => {
-      const wrapper = shallow(
-        <Tabs width="sm" onChange={noop} value={0} variant="scrollable" scrollButtons="auto">
-          <Tab />
-          <Tab />
-        </Tabs>,
-      );
-      assert.strictEqual(wrapper.find(TabScrollButton).length, 2, 'should be zero');
+      assert.strictEqual(wrapper.find(TabScrollButton).length, 2);
       assert.strictEqual(
         wrapper.find(TabScrollButton).everyWhere(node => node.hasClass(classes.scrollButtonsAuto)),
         true,
       );
     });
 
     it('should handle window resize event', () => {
-      const wrapper = shallow(
+      const wrapper = mount(
         <Tabs width="md" onChange={noop} value={0} variant="scrollable" scrollButtons="on">
           <Tab />
           <Tab />
         </Tabs>,
       );
-      const instance = wrapper.instance();
+      const instance = wrapper.find('Tabs').instance();
       stub(instance, 'updateScrollButtonState');
       stub(instance, 'updateIndicatorState');
-      wrapper
-        .find('EventListener')
-        .at(0)
-        .simulate('resize');
+
+      window.dispatchEvent(new window.Event('resize', {}));
```

Comment:
Decouples test from specific implementation. We don't actually care how we listen to the resize event. Just that we listen to window.onResize. This should still pass if we switch to a useEffect and window.addEventListener.